src/main/java/dread/DreadReflectionEnumAnalyzer.java

```diff
@@ -0,0 +1,139 @@
+import ghidra.program.model.symbol.ReferenceManager;
+import ghidra.program.model.symbol.SymbolTable;
+import ghidra.program.util.string.StringSearcher;
+import ghidra.util.exception.CancelledException;
+import ghidra.util.task.TaskMonitor;
+
+public class DreadReflectionEnumAnalyzer extends DreadAnalyzer {
+	public DreadReflectionEnumAnalyzer() {
+		super("(Dread) Generate Enum Classes", "Analyzes functions in order to generate enums", AnalyzerType.FUNCTION_ANALYZER);
+		setPriority(priority(1));
+	}
+	
+	private boolean forceReanalysis = false;
+	
+	@Override
+	public void registerOptions(Options options, Program program) {
+		super.registerOptions(options, program);
+		options.registerOption("Force re-analysis", forceReanalysis, null,
+			"Re-analyze even if a class has already been created");
+	}
+	
+	@Override
+	public void optionsChanged(Options options, Program program) {
+		super.optionsChanged(options, program);
+		forceReanalysis = options.getBoolean("Force re-analysis", forceReanalysis);
+	}
+	
+	private ArrayList<Reference> filterReferenceIterator(ReferenceIterator iterator, RefType type) {
+		ArrayList<Reference> result = new ArrayList<Reference>();
+		for (Reference ref : iterator) {
+			if (ref.getReferenceType() == type) {
+				result.add(ref);
+			}							
+		}
+		return result;
+	}
+	
+	private ArrayList<Reference> filterReferenceIterator(Reference[] iterator, RefType type) {
+		ArrayList<Reference> result = new ArrayList<Reference>();
+		for (Reference ref : iterator) {
+			if (ref.getReferenceType() == type) {
+				result.add(ref);
+			}							
+		}
+		return result;
+	}
+	
+	private String getFirstStringParam(Address initialAddress, ReferenceManager rm, Listing listing) {
+		for (int offset = 4; offset < 0x100; offset += 4) {
+			for (Reference ref : rm.getReferencesFrom(initialAddress.subtract(offset))) {
+				if (ref.getReferenceType() == RefType.PARAM) {
+					Data it = listing.getDataAt(ref.getToAddress());
+					if (it != null && it.getDataType().toString() == "string") {
+						return (String) it.getValue();
+					}				
+				}
+			}
+		}
+		return null;
+	}
+
+	@Override
+	public boolean added(Program program, AddressSetView set, TaskMonitor monitor, MessageLog log)
+			throws CancelledException {
+		
+		FunctionManager fm = program.getFunctionManager();
+		ReferenceManager rm = program.getReferenceManager();
+		Listing listing = program.getListing();
+		final StringSearcher ss = new StringSearcher(program, 0, 1, true, true);
+		SymbolTable st = program.getSymbolTable();
+		
+		Namespace reflection = reflection(program);
+		if (reflection == null) { return false; }
+
+		Function addEnumValue = getRequiredCallees(program).get("AddEnumValue");
+		Set<Function> calling = addEnumValue.getCallingFunctions(null);
+
+		monitor.setProgress(0);
+		monitor.setMaximum(calling.size());
+		monitor.setIndeterminate(false);
+		
+		for (Function f : calling) {
+			monitor.incrementProgress(1);
```

Comment:
add `if (monitor.isCancelled()) { return false; }` so that the user can actually cancel analysis partway through (I only realized this was necessary today)